apps/server/src/modules/task/uc/task.authorization.service.ts

```diff
@@ -17,10 +17,12 @@ export class TaskAuthorizationService {
 	async getPermittedCourses(userId: EntityId, permission: TaskParentPermission): Promise<EntityId[]> {
 		// courseGroups are missing
 		// lessons are missing -> only search for hidden: false,
-		const [permittedCourses] =
-			permission === TaskParentPermission.write
-				? await this.courseRepo.findAllForTeacher(userId)
-				: await this.courseRepo.findAllForStudent(userId);
+		let permittedCourses: Course[] = [];
```

Comment:
can the default case ever occur at all? (given the two possible permissions in `TaskParentPermission`)

Reply:
Yes but at the point where we outsourse authorization it is much cleaner for the person that look into it. It can be done by developers that are not worked on tasks.